crates/signer/src/manager.rs

```diff
@@ -16,125 +23,243 @@ use crate::error::SignerModuleError;
 // with its consensus pubkey When a new commit module starts, pass the
 // ProxyDelegation msg and then sign all future commit messages with the proxy
 // key for slashing the faulty message + proxy delegation can be used
-// Signed using builder domain
+#[derive(Clone, Deref)]
+pub struct BlsProxySigner {
+    #[deref]
+    signer: BlsSigner,
+    delegation: SignedProxyDelegationBls,
+}
+
+#[derive(Clone, Deref)]
+pub struct EcdsaProxySigner {
+    #[deref]
+    signer: EcdsaSigner,
+    delegation: SignedProxyDelegationEcdsa,
+}
+
+#[derive(From)]
+pub enum GenericProxySigner {
```

Comment:
I'd say after losing the generics, the enums `GenericPubkey`, `GenericProxySigner` and the sort around `manager.rs`, remain kind of awkwardly in the middle and don't make much sense. If we decide we're satisfied with splitting up pretty much the entire signer module signing logic into two completely separate flows (one for BLS, one for ECDSA), then we can get rid of these uniting enums, too.

Seems to me like the main purpose they're currently serving is a not so grandiose reduction of duplication within `manager.rs`, which could be a cost worth paying... I'm not sure about the overall split, though.

Reply:
`SignedProxyDelegation` already generic in [a7d53a1](https://github.com/Commit-Boost/commit-boost-client/pull/87/commits/a7d53a15ed13942d361008f28e0f9886516f4083)

Reply:
`SignProxyRequest` also generic in the public key as of [780299e](https://github.com/Commit-Boost/commit-boost-client/pull/87/commits/780299e6b529a96ff9582494868cab559d1b87dc)

This seems like a nice place to stop with the generics - client SDK simplified without any sprawling of generics deeper into the signer module.